File: workspaces/build-python/runner.py

import sys
import json
import importlib.util
import traceback

def run_function(input_data):
    try:
        # Load the function dynamically
        spec = importlib.util.spec_from_file_location("function", "/app/function/function.py")
        module = importlib.util.module_from_spec(spec)
        spec.loader.exec_module(module)
        
        if not hasattr(module, "handler"):
            raise Exception("Function must define a 'handler' function")
        
        # Execute function
        result = module.handler(input_data)
        return result
    except Exception as e:
        return {
            "error": str(e),
            "traceback": traceback.format_exc()
        }

# This file is used by the container to execute functions
# It's not directly exposed to the API but used internally
if __name__ == "__main__":
    import sys
    import os
    import json
    
    # Keep this process running and listen for execution commands
    while True:
        try:
            # This process will be invoked by the Docker engine
            # Input will be passed via stdin or environment variables
            pass
        except Exception as e:
            print(json.dumps({"error": str(e)}))
            sys.exit(1)

File: lambda-serverless/workspaces/build-javascript/runner.js

const fs = require('fs');
const path = require('path');

async function runFunction(input) {
  try {
    // Load the function
    const functionPath = '/app/function/function.js';
    if (!fs.existsSync(functionPath)) {
      throw new Error('Function file not found');
    }
    
    // Clear require cache to ensure we get fresh code
    delete require.cache[require.resolve(functionPath)];
    
    // Import the function
    const fn = require(functionPath);
    
    if (typeof fn.handler !== 'function') {
      throw new Error("Function must export a 'handler' function");
    }
    
    // Execute function
    return await fn.handler(input);
  } catch (error) {
    return {
      error: error.message,
      stack: error.stack
    };
  }
}

// This file is used by the container to execute functions
// It's not directly exposed to the API but used internally
process.on('message', async (message) => {
  try {
    const result = await runFunction(message.input);
    process.send({ success: true, result });
  } catch (error) {
    process.send({ 
      success: false, 
      error: error.message,
      stack: error.stack
    });
  }
});